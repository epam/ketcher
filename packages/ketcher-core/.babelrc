{
  "presets": ["@babel/env", "@babel/typescript"],
  "plugins": ["@babel/plugin-transform-runtime"]
}
