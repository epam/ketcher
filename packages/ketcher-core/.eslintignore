dist/
docs/
node_modules/

# ajv validation file generated with npm run ajv
src/domain/serializers/ket/compiledSchema.js