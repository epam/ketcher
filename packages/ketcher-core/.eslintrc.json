{
  "root": false,
  "extends": "plugin:jest/recommended",
  "rules": {
    "@typescript-eslint/no-non-null-assertion": "warn",
    "no-throw-literal": "warn",
    "@typescript-eslint/no-empty-function": "warn",
    "no-constant-condition": "warn",
    "@typescript-eslint/no-use-before-define": [
      "warn",
      { "functions": false, "typedefs": false }
    ]
  }
}
