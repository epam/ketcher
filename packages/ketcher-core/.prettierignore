dist/
docs/
node_modules/

src/domain/serializers/ket/compiledSchema.js