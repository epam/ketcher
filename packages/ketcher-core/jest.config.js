module.exports = {
  testMatch: ['**/__tests__/**/?(*.)+(spec|test).+(ts|js)'],
  testPathIgnorePatterns: ['fixtures', 'dist', 'node_modules'],
  transform: {
    '\\.js?$': 'babel-jest',
    '^.+\\.(ts|tsx)$': 'ts-jest'
  },
  moduleNameMapper: {
    'application(.*)$': '<rootDir>/src/application/$1',
    'domain(.*)$': '<rootDir>/src/domain/$1',
    'infrastructure(.*)$': '<rootDir>/src/infrastructure/$1',
    'utilities(.*)$': '<rootDir>/src/utilities/$1'
  },
  globals: {
    'ts-jest': {
      diagnostics: {
        warnOnly: true
      }
    }
  }
}
