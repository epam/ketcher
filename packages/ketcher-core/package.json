{
  "name": "ketcher-core",
  "version": "2.12.0-rc.2",
  "description": "Web-based molecule sketcher",
  "license": "Apache-2.0",
  "homepage": "http://lifescience.opensource.epam.com/ketcher",
  "author": {
    "name": "EPAM Life Sciences",
    "url": "http://lifescience.opensource.epam.com"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/epam/ketcher.git",
    "directory": "packages/ketcher-core"
  },
  "keywords": [
    "molecule",
    "sketcher",
    "chemistry"
  ],
  "bugs": {
    "url": "https://github.com/epam/ketcher/issues"
  },
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "module": "dist/index.modern.js",
  "source": "src/index.ts",
  "engines": {
    "node": ">=14"
  },
  "scripts": {
    "build": "cross-env NODE_ENV=production rollup -c -m true",
    "start": "cross-env NODE_ENV=development rollup -c -m true -w",
    "test": "run-s test:prettier test:eslint:quiet test:types test:unit",
    "test:eslint": "eslint . --ext .ts,.js",
    "test:eslint:quiet": "npm run test:eslint -- --quiet",
    "test:eslint:fix": "npm run test:eslint -- --fix",
    "test:types": "tsc --noEmit",
    "test:unit": "jest",
    "test:prettier": "prettier --check \"./**/*.{js,ts,json}\"",
    "prettier:write": "prettier --write \"./**/*.{js,ts,json}\"",
    "typedoc": "typedoc"
  },
  "dependencies": {
    "@babel/runtime": "^7.17.9",
    "ajv": "^8.10.0",
    "assert": "^2.0.0",
    "lodash": "^4.17.21",
    "raphael": "^2.3.0",
    "svgpath": "^2.3.1"
  },
  "devDependencies": {
    "@babel/core": "^7.17.9",
    "@babel/plugin-transform-runtime": "^7.17.0",
    "@babel/preset-env": "^7.16.11",
    "@babel/preset-typescript": "^7.16.7",
    "@rollup/plugin-babel": "^5.2.1",
    "@rollup/plugin-commonjs": "^16.0.0",
    "@rollup/plugin-json": "^4.1.0",
    "@rollup/plugin-node-resolve": "^10.0.0",
    "@rollup/plugin-replace": "^2.3.4",
    "@rollup/plugin-strip": "^2.0.0",
    "@types/jest": "^27.0.3",
    "@types/node": "^16.11.12",
    "@zerollup/ts-transform-paths": "^1.7.18",
    "babel-jest": "26.6.0",
    "cross-env": "^7.0.3",
    "eslint": "^8.44.0",
    "eslint-plugin-jest": "^25.3.0",
    "jest": "26.6.0",
    "jest-mock-extended": "^2.0.4",
    "npm-run-all": "^4.1.5",
    "prettier": "^2.5.1",
    "rollup": "^2.60.2",
    "rollup-plugin-cleanup": "^3.2.1",
    "rollup-plugin-copy": "^3.4.0",
    "rollup-plugin-delete": "^2.0.0",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-typescript2": "^0.31.1",
    "rollup-plugin-visualizer": "^5.5.2",
    "ts-jest": "26.5.6",
    "ttypescript": "^1.5.13",
    "typedoc": "^0.22.17",
    "typedoc-plugin-markdown": "^3.12.1",
    "typescript": "^4.5.2"
  },
  "files": [
    "dist"
  ]
}
