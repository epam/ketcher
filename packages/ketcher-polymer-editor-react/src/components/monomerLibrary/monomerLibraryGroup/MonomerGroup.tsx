/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/
import { MonomerItem, MonomerItemType } from '../monomerLibraryItem'
import styled from '@emotion/styled'

interface MonomerGroupProps {
  items: MonomerItemType[]
  onItemClick: (item: MonomerItemType) => void
  title?: string
}

const ItemsContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  flex-flow: row wrap;
  gap: 8px;
  flex: 1;
  justify-content: space-between;
  margin-bottom: 24px;

  &::after {
    content: '';
    flex: auto;
  }
`

const GroupContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
  font-size: ${({ theme }) => theme.ketcher.font.size.small};
  font-family: ${({ theme }) => theme.ketcher.font.family.roboto};
  color: ${({ theme }) => theme.ketcher.color.divider};
  margin: 0;
  gap: 8px;
`

const GroupTitle = styled.div`
  height: 100%;
  display: flex;
  flex-grow: 0;
  flex-basis: 14px;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: flex-start;
  font-size: ${({ theme }) => theme.ketcher.font.size.medium};
  font-family: ${({ theme }) => theme.ketcher.font.family.roboto};
  color: ${({ theme }) => theme.ketcher.color.text.primary};
  margin: 0;
`

const MonomerGroup = (props: MonomerGroupProps) => {
  const { items, title, onItemClick } = props

  return (
    <GroupContainer>
      {title && (
        <GroupTitle>
          <span>{title}</span>
        </GroupTitle>
      )}
      <ItemsContainer>
        {items.map((monomer) => {
          const key = monomer.props
            ? `${monomer.props.MonomerName + monomer.props.Name}`
            : monomer.label
          return (
            <MonomerItem
              key={key}
              item={monomer}
              onClick={() => onItemClick(monomer)}
            />
          )
        })}
      </ItemsContainer>
    </GroupContainer>
  )
}
export { MonomerGroup }
