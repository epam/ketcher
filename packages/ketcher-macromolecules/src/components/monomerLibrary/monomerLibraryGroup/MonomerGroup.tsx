/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/
import { useCallback, useEffect, useMemo, useState } from 'react';
import { EmptyFunction } from 'helpers';
import { debounce } from 'lodash';
import { MonomerItem } from '../monomerLibraryItem';
import { GroupContainer, GroupTitle, ItemsContainer } from './styles';
import { IMonomerGroupProps } from './types';
import { getMonomerUniqueKey } from 'state/library';
import { MonomerItemType } from 'ketcher-core';
import { calculatePreviewPosition } from '../../../helpers';
import { useAppDispatch, useAppSelector } from 'hooks';
import {
  showPreview,
  selectShowPreview,
  selectEditor,
  selectTool,
} from 'state/common';
import { selectActiveRnaBuilderItem } from 'state/rna-builder';

const MonomerGroup = ({
  items,
  title,
  selectedMonomerUniqueKey,
  libraryName,
  disabled,
  onItemClick = EmptyFunction,
}: IMonomerGroupProps) => {
  const dispatch = useAppDispatch();
  const preview = useAppSelector(selectShowPreview);
  const editor = useAppSelector(selectEditor);
  const activeMonomerGroup = useAppSelector(selectActiveRnaBuilderItem);

  const [selectedItemInGroup, setSelectedItemInGroup] =
    useState<MonomerItemType | null>(null);

  useEffect(() => {
    setSelectedItemInGroup(null);
  }, [activeMonomerGroup]);

  const dispatchShowPreview = useCallback(
    (payload) => dispatch(showPreview(payload)),
    [dispatch],
  );

  const debouncedShowPreview = useMemo(
    () => debounce((p) => dispatchShowPreview(p), 500),
    [dispatchShowPreview],
  );

  const handleItemMouseLeave = () => {
    debouncedShowPreview.cancel();
    dispatch(showPreview(undefined));
  };

  const handleItemMouseMove = (
    monomer: MonomerItemType,
    e: React.MouseEvent<HTMLDivElement, MouseEvent>,
  ) => {
    handleItemMouseLeave();
    if (preview.monomer || !e.currentTarget) {
      return;
    }
    const cardCoordinates = e.currentTarget.getBoundingClientRect();
    const previewStyle = calculatePreviewPosition(monomer, cardCoordinates);
    const style = { top: previewStyle, right: '-88px' };
    debouncedShowPreview({ monomer, style });
  };

  const selectMonomer = (monomer: MonomerItemType) => {
    dispatch(selectTool('monomer'));
    setSelectedItemInGroup(monomer);

    if (['FAVORITES', 'PEPTIDE', 'CHEM'].includes(libraryName ?? '')) {
      editor.events.selectMonomer.dispatch(monomer);
    }

    onItemClick(monomer);
  };

  const isMonomerSelected = (monomer: MonomerItemType) => {
    return selectedItemInGroup
      ? getMonomerUniqueKey(selectedItemInGroup) ===
          getMonomerUniqueKey(monomer)
      : selectedMonomerUniqueKey === getMonomerUniqueKey(monomer);
  };

  return (
    <GroupContainer>
      {title && (
        <GroupTitle>
          <span>{title}</span>
        </GroupTitle>
      )}
      <ItemsContainer>
        {items.map((monomer) => {
          const key = monomer.props
            ? `${monomer.props.MonomerName + monomer.props.Name}`
            : monomer.label;
          return (
            <MonomerItem
              key={key}
              disabled={disabled}
              item={monomer}
              isSelected={isMonomerSelected(monomer)}
              onMouseLeave={handleItemMouseLeave}
              onMouseMove={(e) => handleItemMouseMove(monomer, e)}
              onClick={() => selectMonomer(monomer)}
            />
          );
        })}
      </ItemsContainer>
    </GroupContainer>
  );
};
export { MonomerGroup };
