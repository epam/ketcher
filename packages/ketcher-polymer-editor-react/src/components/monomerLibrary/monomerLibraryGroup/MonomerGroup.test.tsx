/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import { render, screen, fireEvent } from '@testing-library/react';
import { Struct } from 'ketcher-core';

import { MonomerGroup } from './MonomerGroup';

describe('Monomer Group', () => {
  const mockGroupProps = {
    groupItems: [
      {
        label: 'R',
        props: {
          MonomerNaturalAnalogCode: 'R',
          MonomerName: 'RRMonomerName',
          Name: 'RRName',
        },
        struct: new Struct(),
      },
      {
        label: 'm',
        props: {
          MonomerNaturalAnalogCode: 'A',
          MonomerName: 'mAMonomerName',
          Name: 'mAName',
        },
        struct: new Struct(),
      },
      {
        label: 'd',
        props: {
          MonomerNaturalAnalogCode: 'D',
          MonomerName: 'dDMonomerName',
          Name: 'dDName',
        },
        struct: new Struct(),
      },
      {
        label: 'ar',
        props: {
          MonomerNaturalAnalogCode: 'R',
          MonomerName: 'arRMonomerName',
          Name: 'arRName',
        },
        struct: new Struct(),
      },
      {
        label: 'Ld',
        props: {
          MonomerNaturalAnalogCode: 'L',
          MonomerName: 'LdLMonomerName',
          Name: 'LdLRName',
        },
        struct: new Struct(),
      },
    ],
    groupTitle: 'Mock title',
  };
  const onItemClick = jest.fn();

  it('should render correct without title prop', () => {
    const view = render(
      withThemeAndStoreProvider(
        <MonomerGroup
          items={mockGroupProps.groupItems}
          onItemClick={onItemClick}
        />,
      ),
    );

    mockGroupProps.groupItems.forEach((item) => {
      const currentItem = screen.getByText(item.label);
      expect(currentItem).toBeInTheDocument();
    });

    expect(view).toMatchSnapshot();
  });
  it('should render correct with title prop', () => {
    const view = render(
      withThemeAndStoreProvider(
        <MonomerGroup
          items={mockGroupProps.groupItems}
          title={mockGroupProps.groupTitle}
          onItemClick={onItemClick}
        />,
      ),
    );

    const title = screen.getByText(mockGroupProps.groupTitle);
    expect(title).toBeInTheDocument();

    expect(view).toMatchSnapshot();
  });
  it('callback for monomer item should be called for group items', () => {
    render(
      withThemeAndStoreProvider(
        <MonomerGroup
          items={mockGroupProps.groupItems}
          title={mockGroupProps.groupTitle}
          onItemClick={onItemClick}
        />,
      ),
    );
    const item = screen.getByText('Ld');
    fireEvent.click(item);
    expect(onItemClick).toBeCalled();
  });
});
