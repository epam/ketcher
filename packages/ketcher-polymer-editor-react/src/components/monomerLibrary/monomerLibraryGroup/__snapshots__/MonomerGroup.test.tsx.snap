// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monomer Group should render correct with title prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1jn33hb"
      >
        <div
          class="css-a0lsh8"
        >
          <span>
            Mock title
          </span>
        </div>
        <div
          class="css-u8jo8f"
        >
          <div
            class="css-ryv5uk"
            data-testid="R"
          >
            <span>
              R
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
          <div
            class="css-17atjlm"
            data-testid="A"
          >
            <span>
              m
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
          <div
            class="css-kvk3yi"
            data-testid="D"
          >
            <span>
              d
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
          <div
            class="css-ryv5uk"
            data-testid="R"
          >
            <span>
              ar
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
          <div
            class="css-1tniqtk"
            data-testid="L"
          >
            <span>
              Ld
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1jn33hb"
    >
      <div
        class="css-a0lsh8"
      >
        <span>
          Mock title
        </span>
      </div>
      <div
        class="css-u8jo8f"
      >
        <div
          class="css-ryv5uk"
          data-testid="R"
        >
          <span>
            R
          </span>
          <div
            class="star "
          >
            ★
          </div>
        </div>
        <div
          class="css-17atjlm"
          data-testid="A"
        >
          <span>
            m
          </span>
          <div
            class="star "
          >
            ★
          </div>
        </div>
        <div
          class="css-kvk3yi"
          data-testid="D"
        >
          <span>
            d
          </span>
          <div
            class="star "
          >
            ★
          </div>
        </div>
        <div
          class="css-ryv5uk"
          data-testid="R"
        >
          <span>
            ar
          </span>
          <div
            class="star "
          >
            ★
          </div>
        </div>
        <div
          class="css-1tniqtk"
          data-testid="L"
        >
          <span>
            Ld
          </span>
          <div
            class="star "
          >
            ★
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Monomer Group should render correct without title prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1jn33hb"
      >
        <div
          class="css-u8jo8f"
        >
          <div
            class="css-ryv5uk"
            data-testid="R"
          >
            <span>
              R
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
          <div
            class="css-17atjlm"
            data-testid="A"
          >
            <span>
              m
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
          <div
            class="css-kvk3yi"
            data-testid="D"
          >
            <span>
              d
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
          <div
            class="css-ryv5uk"
            data-testid="R"
          >
            <span>
              ar
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
          <div
            class="css-1tniqtk"
            data-testid="L"
          >
            <span>
              Ld
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1jn33hb"
    >
      <div
        class="css-u8jo8f"
      >
        <div
          class="css-ryv5uk"
          data-testid="R"
        >
          <span>
            R
          </span>
          <div
            class="star "
          >
            ★
          </div>
        </div>
        <div
          class="css-17atjlm"
          data-testid="A"
        >
          <span>
            m
          </span>
          <div
            class="star "
          >
            ★
          </div>
        </div>
        <div
          class="css-kvk3yi"
          data-testid="D"
        >
          <span>
            d
          </span>
          <div
            class="star "
          >
            ★
          </div>
        </div>
        <div
          class="css-ryv5uk"
          data-testid="R"
        >
          <span>
            ar
          </span>
          <div
            class="star "
          >
            ★
          </div>
        </div>
        <div
          class="css-1tniqtk"
          data-testid="L"
        >
          <span>
            Ld
          </span>
          <div
            class="star "
          >
            ★
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
